### PR TITLE
Feature: Added audio bitrate to the Details Pane

### LINKED ISSUE #15282: Feature: Add audio file bitrate to details pane
### What feature or improvement do you think would benefit Files?

In file explorer the bitrate for audio files, along with other things like title, contributing artists, album name and length are previewed in the details tab.  But on files app it shows all that except for bitrate.  Instead you have to scroll down and click on 'properties', then 'details', and it will show the bitrate. For me and maybe other people, I sort and download a lot of audio files and its very helpful to be able to see the bitrate at a glance, so it would be nice to have it there next to all the other details.

Sidenote - on file explorer the bitrate is slightly different than what is shown on files app, I'm not sure if this is a bug or its just read differently. For example a 320kbps audio file on file explorer is shown as 312.5 kbps on files app, and a 192kbps file is shown as 187.5 on files app.  Im not sure why

image
image

### Requirements

- This proposal will accomplish ease of access for people who frequently check the bitrate of audio files

### Files Version

3.3.0.0

### Windows Version

23H2

### Comments

_No response_

### PR DESCRIPTION
**Pull request by Artoryn**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #15282

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [x] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
     - Small consolidation of formatting of the bitrate display string. This was done as the same formatting was going to be added in a third location so centralizing it and using the DisplayFunctions already present would make more sense.
     - Change the bitrate from 1024 for the conversion to 1000 (bytes to bits), as this would fix the "Sidenote" that the issue mentioned. This was done to make sure the bitrate is calculated off bits and not bytes better matching file explorer's display of this value.
- [x] Are there any other steps that were used to validate these changes?
>   1. Open app (Audio validation)
>   2. Navigated to directory with a .mp3, .mp4, and .wav audio file
>   3. Verified that the Bitrate showed on the Preview Pane Detail window
>   4. Opened individual file properties window
>   5. Navigated to the Details section
>   6. Verified that the Bitrate still showed properly formatted under the properties window 
>
> ---
>   1. Open app (Video validation)
>   2. Navigated to a directory with a .mp4 video file
>   3. Verified that the Bitrate showed on the Preview Pane Detail window
>     a. Added string title text of "Audio Encoding Bitrate" as it seemed unclear for video files which in properties has 2 "Encoding Bitrate" tags under Audio and Video
>   4. Opened individual file properties window
>   5. Navigated to the Details section
>   6. Verified that the Bitrate for Audio and Video still showed properly formatted under the properties window
>
> ---
>   1. Open app (Combined file validation different bitrates)
>   2. Navigated to a directory with multiple video/audio files
>   3. Verified that the Bitrate showed on the Preview Pane Detail window for individual files
>   4. Selected multiple files (with different bitrates)
>   5. Opened properties window
>   6. Navigated to the Details section
>   7. Verified that "(multiple values)" displays under bitrate
>
> ---
>   1. Open app (Combined file validation same bitrates)
>   2. Navigated to a directory with video/audio file
>   3. Verified that the Bitrate showed on the Preview Pane Detail window for individual file
>   4. Create a copy of a file for testing
>   4. Selected both files (original and copy of file)
>   5. Opened properties window
>   6. Navigated to the Details section
>   7. Verified that the expected Bitrate displays under bitrate

**Screenshots (optional)**
image

### CODE REVIEW

**Review by 0x5bfa — APPROVED** [review]

**Comment by hishitetsu** [comment]
@Artoryn is this still a work in progress?

**Comment by Artoryn** [comment]
> @Artoryn is this still a work in progress?

No, as long as the few refactors to existing functionality are fine, it is good to go. I marked it Ready for Review.

**Review by 0x5bfa — APPROVED** [review]
Sorry for the last stale review.

The code is pretty looking great. LGTM.

File: src/Files.App/ViewModels/Properties/Items/FileProperty.cs [hunk 1]
```diff
@@ -363,9 +363,22 @@ public async static Task<List<FileProperty>> RetrieveAndInitializePropertiesAsyn
...
 		private static string TimeSpanToString(TimeSpan t)
 			=> t.Days > 0 ? (t.Days * 24 + t.Hours) + t.ToString("':'mm':'ss") : t.ToString("hh':'mm':'ss");
+
+		private static string FormatEncodingBitrate(object input)
+		{
+			// For cases when multiple files are selected and it has a string value
+			if (input.GetType() != typeof(uint))
+				return input?.ToString() ?? string.Empty;
+
+			var sizes = new string[] { "bps", "kbps", "mbps", "gbps" };
```
Comment (hishitetsu) [inline src/Files.App/ViewModels/Properties/Items/FileProperty.cs:378]:
```suggestion
			var sizes = new string[] { "bps", "kbps", "Mbps", "Gbps" };
```
Mega and Giga should be capitalized.

**Comment by hishitetsu** [comment]
@yaira2 what about also adding video encoding bitrate to video files?

**Comment by yaira2** [comment]
> @yaira2 what about also adding video encoding bitrate to video files?

Good idea, I'm on board with this.

**Comment by Artoryn** [comment]
This was the original reason I added the new text for specifying Audio over just the Encoding Bitrate as well.
image

Will get these changes pushed.

File: src/Files.App/Strings/en-US/Resources.resw [hunk 2]
```diff
@@ -882,6 +882,12 @@
   <data name="PropertyEncodingBitrate" xml:space="preserve">
     <value>Encoding Bitrate</value>
   </data>
+  <data name="PropertyPreviewAudioEncodingBitrate" xml:space="preserve">
+	<value>Audio Encoding Bitrate</value>
```
Comment (hishitetsu) [inline src/Files.App/Strings/en-US/Resources.resw:886]:
```suggestion
    <value>Audio Encoding Bitrate</value>
```
An indentation issue.

**Review by hishitetsu — APPROVED** [review]
LGTM 👍

**Comment by yaira2** [comment]
@Artoryn thank you for contributing to Files.

**Pull request merged**